lib/macho/headers.rb

```diff
@@ -61,17 +69,69 @@ module MachO
   # @see http://llvm.org/docs/doxygen/html/Support_2MachO_8h_source.html
   CPU_SUBTYPE_LIB64 = 0x80000000
 
-  # all x86-type CPUs
+  # all CPU sub-types for CPU type `CPU_TYPE_I386`
   CPU_SUBTYPE_X86_ALL = 3
 
-  # all x86-type CPUs (what makes this different from CPU_SUBTYPE_X86_ALL?)
-  CPU_SUBTYPE_X86_ARCH1 = 4
+  # all CPU sub-types for CPU type `CPU_TYPE_X86_64`
+  CPU_SUBTYPE_X86_64_ALL = CPU_SUBTYPE_X86_ALL
+
+  # all CPU sub-types for CPU type `CPU_TYPE_X86_64` (Haskell tree)
+  CPU_SUBTYPE_X86_64_H = 8
+
+  # all CPU sub-types for CPU type `CPU_TYPE_ARM`
+  CPU_SUBTYPE_ARM_V7 = 9
+  CPU_SUBTYPE_ARM_V7S = 11
+  CPU_SUBTYPE_ARM_V7K = 12
+
+  # all CPU sub-types for CPU-type `CPU_TYPE_ARM64`
+  CPU_SUBTYPE_ARM64_ALL = 0
+
+  # PowerPC G3
+  CPU_SUBTYPE_POWERPC_750 = 9
 
-  # association of cpu subtypes to string representations
+  # PowerPC G4
+  CPU_SUBTYPE_POWERPC_7400 = 10
+
+  # PowerPC G4 "Voyager"
+  CPU_SUBTYPE_POWERPC_7450 = 11
+
+  # PowerPC G5
+  CPU_SUBTYPE_POWERPC_970 = 100
+
+  # all CPU sub-types for CPU type `CPU_TYPE_POWERPC`
+  CPU_SUBTYPE_POWERPC_ALL = 0
+
+  # all CPU sub-types for CPU type `CPU_TYPE_POWERPC64`
+  CPU_SUBTYPE_POWERPC64_ALL = CPU_SUBTYPE_POWERPC_ALL
+
+  # association of CPU types/subtype pairs to symbol representations
   CPU_SUBTYPES = {
-    CPU_SUBTYPE_X86_ALL => "CPU_SUBTYPE_X86_ALL",
-    CPU_SUBTYPE_X86_ARCH1 => "CPU_SUBTYPE_X86_ARCH1"
-  }
+    CPU_TYPE_I386 => {
+      CPU_SUBTYPE_X86_ALL => :i386,
+    }.freeze,
+    CPU_TYPE_X86_64 => {
+      CPU_SUBTYPE_X86_64_ALL => :x86_64,
+      CPU_SUBTYPE_X86_64_H => :x86_64h,
+    }.freeze,
+    CPU_TYPE_ARM => {
+      CPU_SUBTYPE_ARM_V7 => :armv7,
+      CPU_SUBTYPE_ARM_V7S => :armv7s,
+      CPU_SUBTYPE_ARM_V7K => :armv7k,
+    }.freeze,
+    CPU_TYPE_ARM64 => {
+      CPU_SUBTYPE_ARM64_ALL => :arm64,
+    }.freeze,
+    CPU_TYPE_POWERPC => {
+      CPU_SUBTYPE_POWERPC_750 => :ppc750,
+      CPU_SUBTYPE_POWERPC_7400 => :ppc7400,
+      CPU_SUBTYPE_POWERPC_7450 => :ppc7450,
+      CPU_SUBTYPE_POWERPC_ALL => :ppc,
+    }.freeze,
+    CPU_TYPE_POWERPC64 => {
+      CPU_SUBTYPE_POWERPC_970 => :ppc970,
```

Comment:
I think you're a bit confused here. The PowerPC 970 CPU supported both 32-bit and 64-bit binaries.

32-bit binaries were, if they relied on CPU instruction unavailable prior to the 970 got a type of `CPU_TYPE_POWERPC` and a sub-type of `CPU_SUBTYPE_POWERPC_970`. But the 970 was still perfectly capable of executing 32-bit binaries with other (earlier introduced) sub-types.

For 64-bit binaries a type of `CPU_TYPE_POWERPC64` and sub-type of `CPU_SUBTYPE_POWERPC64_ALL` effectively means PowerPC 970, because this was the first 64-bit PowerPC processor. (If there had been any successors, they most likely would have been compatible to the 970 and just extended its instruction set, possibly making other `CPU_SUBTYPE_POWERPC64_*` sub-types necessary.)